Documentation/cmdref/cilium-agent.md

```diff
@@ -210,6 +210,7 @@ cilium-agent [flags]
       --hubble-drop-events                                        Emit packet drop Events related to pods (alpha)
```

Comment:
One bit of feedback I have is how large this commit is, could you see if you could break this down into several smaller commits to make review, diffing and backporting easier?

See: [Cilium contribution docs section under how to submit a pull request.](https://docs.cilium.io/en/latest/contributing/development/contributing_guide/#submitting-a-pull-request)

Reply:
@rectified95 

There's no hard and fast rule for commit size, but 1700 anything that's not all generated/vendored code changes does seem like a lot.  Specifically, I worry this might make identifying changes that potentially caused regressions more difficult.

Here's how I usually approach this (specifically, in the case of a PR that introduces new functionality):

* Flag & generated CLI documentation changes can be in their own commit.
* Put helm changes in their own commit.
* Primarily new code, such as that in `pkg/hubble/metrics/metric_config_watcher.go` and `pkg/hubble/metrics/dynamic_flow_processor.go` and related test changes can probably be in a seperate commit while not breaking the per-commit build rule.
* Finally, in-line changes integrating the new code into existing code can be their own commits.  As a reviewer, I find these ones particularly helpful as the commit message and smaller set of changes can make it easier to understand the broader impact of the change.

Reply:
@tommyp1ckles Per your suggestion, I've split this PR into several commits: docs, Helm, new code, integration code.
To ensure each one builds, I added empty implementations of the new interface methods in the intermediate commits.
Since this operation undoes earlier squashing, it was at times tricky - I ended up putting the test code with the integration commit since it relies on its final shape post-refactoring. 

Changelist for the integration commit (here, since I've seen the CI reject commit messages over 130 characters.. unless that was just the title line):
- fork the metrics setup path in Hubble launch to use the dynamic config path
- remove the `FlowProcessor` interface and move its `ProcessFlow` method into the `Handler` interface (rationale in PR description + this thread: https://github.com/cilium/cilium/pull/35185#discussion_r1797444382)
- remove the wrapper type `Handlers` now that we don't maintain the list of `FlowProcessor`s
- convert handler bulk creation functions to operate on a single handler and be iterated on (needed for DFP applying metric configs one by one)
- adapt existing tests to use refactored types

The resulting diff from the force-push contains 3 changes:
- empty line removal 
- removed commented import line
- removed redundncy I spotted in DFP: it already creates the map it was calling a helper for

https://github.com/cilium/cilium/compare/0658165559d45752ebaa6310d990bb43be1ebe8b..734d5a8fd6dd1f87379fb9fbd582a3114ccb3399